x/stakeibc/keeper/update_validator_shares_exch_rate_test.go

```diff
@@ -0,0 +1,236 @@
+}
+
+// ================================== 2: QueryDelegationsIcq ==========================================
+
+type QueryDelegationsIcqTestCase struct {
+	hostZone           types.HostZone
+	valoperAddr        string
+	strideEpochTracker types.EpochTracker
+	dayEpochTracker    types.EpochTracker
+}
+
+func (s *KeeperTestSuite) SetupQueryDelegationsIcq() QueryDelegationsIcqTestCase {
+	currentEpoch := uint64(1)
+	valoperAddr := "cosmosvaloper133lfs9gcpxqj6er3kx605e3v9lqp2pg5syhvsz"
+
+	// set up IBC
+	s.CreateTransferChannel(HostChainId)
+
+	delegationAccountOwner := fmt.Sprintf("%s.%s", HostChainId, "DELEGATION")
+	s.CreateICAChannel(delegationAccountOwner)
+	delegationAddress := s.IcaAddresses[delegationAccountOwner]
+
+	hostZone := types.HostZone{
+		ChainId:           HostChainId,
+		HostDenom:         Atom,
+		IBCDenom:          IbcAtom,
+		Bech32Prefix:      Bech32Prefix,
+		DelegationAccount: &stakeibctypes.ICAAccount{Address: delegationAddress},
+	}
+
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx(), hostZone)
+
+	// This will make the current time 90% through the epoch
+	strideEpochTracker := types.EpochTracker{
+		EpochIdentifier:    epochtypes.STRIDE_EPOCH,
+		EpochNumber:        currentEpoch,
+		Duration:           10_000_000_000,                                               // 10 second epochs
+		NextEpochStartTime: uint64(s.Coordinator.CurrentTime.UnixNano() + 1_000_000_000), // epoch ends in 1 second
+	}
+	s.App.StakeibcKeeper.SetEpochTracker(s.Ctx(), strideEpochTracker)
+
+	// This will make the current time 50% through the day
+	dayEpochTracker := types.EpochTracker{
+		EpochIdentifier:    epochtypes.DAY_EPOCH,
+		EpochNumber:        currentEpoch,
+		Duration:           40_000_000_000,                                                // 40 second epochs
+		NextEpochStartTime: uint64(s.Coordinator.CurrentTime.UnixNano() + 20_000_000_000), // day ends in 20 second
+	}
+	s.App.StakeibcKeeper.SetEpochTracker(s.Ctx(), dayEpochTracker)
+
+	return QueryDelegationsIcqTestCase{
+		hostZone:           hostZone,
+		valoperAddr:        valoperAddr,
+		strideEpochTracker: strideEpochTracker,
+		dayEpochTracker:    dayEpochTracker,
+	}
+}
+
+func (s *KeeperTestSuite) TestQueryDelegationsIcq_Successful() {
+	tc := s.SetupQueryDelegationsIcq()
+
+	err := s.App.StakeibcKeeper.QueryDelegationsIcq(s.Ctx(), tc.hostZone, tc.valoperAddr)
+	s.Require().NoError(err, "no error expected")
+
+	// check a query was created (a simple test; details about queries are covered in makeRequest's test)
+	queries := s.App.InterchainqueryKeeper.AllQueries(s.Ctx())
+	s.Require().Len(queries, 1, "one query should have been created")
+
+	// querying twice with the same query should only create one query
+	err = s.App.StakeibcKeeper.QueryDelegationsIcq(s.Ctx(), tc.hostZone, tc.valoperAddr)
+	s.Require().NoError(err, "no error expected")
+
+	// check a query was created (a simple test; details about queries are covered in makeRequest's test)
```

Comment:
nit: rm

Reply:
I actually like this for clarity, that's why I included it

Reply:
otherwise unclear what's being tested IMO

Reply:
I think the comment is stale though? Cause a second query should not be created

Reply:
reason I was saying to remove is cause there's a comment 3 lines up that says what's being tested. But rewording it is also fine!